From 0e5c3591c48ddab2d3701462b3a4a9685c32f1ee Mon Sep 17 00:00:00 2001
From: Quentin Young <qlyoung@nvidia.com>
Date: Tue, 29 Mar 2022 16:16:35 -0400
Subject: [PATCH] tests: skip grpc test if frr not installed

it wants yang models installed which will only be there if frr has been
installed before, causing `make check` to fail when run on a system on
which frr has not been installed when GRPC is enabled (--enable-grpc)

Signed-off-by: Quentin Young <qlyoung@nvidia.com>
[ Changed check from /usr/share/yang existence - which is created by libyang
  already - to check specifically if any frr yang files exist. ]
Signed-off-by: Brennan Lamoreaux <blamoreaux@vmware.com>
---
 tests/lib/test_grpc.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tests/lib/test_grpc.py b/tests/lib/test_grpc.py
index 06ae6c0..d9f15dd 100644
--- a/tests/lib/test_grpc.py
+++ b/tests/lib/test_grpc.py
@@ -3,6 +3,7 @@ import os
 import subprocess
 import pytest
 import frrtest
+import glob

 class TestGRPC(object):
     program = "./test_grpc"
@@ -11,6 +12,10 @@ class TestGRPC(object):
         'S["GRPC_TRUE"]=""\n' not in open("../config.status").readlines(),
         reason="GRPC not enabled",
     )
+    @pytest.mark.skipif(
+        not glob.glob("/usr/share/yang/frr*.yang"),
+        reason="FRR Yang models aren't installed in /usr/share/yang",
+    )
     def test_exits_cleanly(self):
         basedir = os.path.dirname(inspect.getsourcefile(type(self)))
         program = os.path.join(basedir, self.program)
--
2.39.0

